src/main/java/nextstep/fp/Lambda.java

```diff
@@ -27,30 +27,18 @@ public void run() {
     }
 
     public static int sumAll(List<Integer> numbers) {
-        int total = 0;
-        for (int number : numbers) {
-            total += number;
-        }
-        return total;
+        return sum(numbers, number -> true);
     }
 
     public static int sumAllEven(List<Integer> numbers) {
-        int total = 0;
-        for (int number : numbers) {
-            if (number % 2 == 0) {
-                total += number;
-            }
-        }
-        return total;
+        return sum(numbers, number -> number % 2 == 0);
     }
 
     public static int sumAllOverThree(List<Integer> numbers) {
-        int total = 0;
-        for (int number : numbers) {
-            if (number > 3) {
-                total += number;
-            }
-        }
-        return total;
+        return sum(numbers, number -> number > 3);
+    }
+
+    private static int sum(List<Integer> numbers, Conditional conditional) {
+        return numbers.stream().filter(conditional::test).reduce(0, Integer::sum);
     }
```

Comment:
👍

Reply:
```suggestion
    private static int sum(List<Integer> numbers, Conditional conditional) {
        return numbers.stream().filter(conditional::test).mapToInt(it -> it).sum();
    }
```

Integer를 값 객체로 래핑했다면 reduce도 좋을 것 같네요
지금은 간단한 정수 연산이기 때문에 IntStream을 사용하면 sum을 쉽게 구할 수 있습니다.